extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -720,17 +720,27 @@ export async function fileContainsCreateTableStatement(fullPath: string, project
  * @param serverInfo server information
  * @returns target platform for the database project
  */
-export async function getTargetPlatformFromServerVersion(serverInfo: azdataType.ServerInfo | vscodeMssql.IServerInfo): Promise<SqlTargetPlatform | undefined> {
+export async function getTargetPlatformFromServerVersion(serverInfo: azdataType.ServerInfo | vscodeMssql.IServerInfo, serverName?: string): Promise<SqlTargetPlatform | undefined> {
```

Comment:
Adding the server URL here as a separate parameter from serverInfo feels odd/messy.  Is there a reason this can't be exposed in the ServerInfo objects?

Reply:
Server name is readily available in the vscode API but not via azdata. From what I could find, server name is only available from the connection profile in azdata.

Reply:
I was thinking that maybe it wouldn't be too much to add it as a new property to the ServerInfo object that either gets back from STS, but that's probably dreaming here.  I'm shocked that it's not present already in that object.

Feel free to just resolve this, then.  I may go back and add it to that struct later to clean this up.

Reply:
Thanks Benjin

Reply:
Actually using the server name from connection profile doesn't work for all scenarios. It only works via right-click on an open connection -> Create Database Project. This doesn't work when clicking "Create Project from Database" via the Projects extension. I think we need some changes in the azdata API for this: #24202